testing/sdk_auth.py

```diff
@@ -0,0 +1,383 @@
+        for node in nodes:
+            if node["id"] == host_id:
+                log.info("Host name is {host_name}".format(host_name=node["hostname"]))
+                return node["hostname"]
+
+    raise RuntimeError("Failed to get name of host running the KDC app: {nodes}")
+
+
+@retry(stop_max_attempt_number=2, wait_fixed=1000)
+def _get_master_public_ip() -> str:
+    """
+    :return (str): The public IP of the master node in the DC/OS cluster.
+    """
+    dcos_url, headers = sdk_security.get_dcos_credentials()
+    cluster_metadata_url = "{cluster_url}/metadata".format(cluster_url=dcos_url)
+    request_metadata = {
+            "--header": "Authorization: {token}".format(token=headers["Authorization"])
+    }
+    raw_response = sdk_cmd.request("GET", cluster_metadata_url, request_metadata, verify=False)
+    if not raw_response.ok:
+        raise RuntimeError("Unable to get the master node's public IP address: {err}".format(err=repr(response)))
+
+    response = json.loads(raw_response.text)
+    if "PUBLIC_IPV4" not in response:
+        raise KeyError("Cluster metadata does not include master's public ip: {response}".format(
+            response=repr(raw_response)))
+
+    public_ip = response["PUBLIC_IPV4"]
+    log.info("Master public ip is {public_ip}".format(public_ip=public_ip))
+    return public_ip
+
+
+def _create_temp_working_dir() -> str:
+    """
+    Creates a temporary working directory to enable setup of the Kerberos environment.
+    :return (str): The path of the working directory.
+    """
+    dir_name = "/tmp/{kerberos}_{current_time}".format(kerberos=KERBEROS_APP_ID, current_time=str(time.time()))
+    log.info("Creating temp working directory {}".format(dir_name))
+    try:
+        os.makedirs(dir_name)
+    except OSError as e:
+        raise OSError("Can't create temp working directory: {}".format(repr(e)))
+
+    return dir_name
+
+
+def _copy_file_to_localhost(self):
+    """
+    Copies the keytab that was generated inside the container running the KDC server to the localhost
+    so it can be uploaded to the secret store later. This must be done in multiple steps as we have to jump through
+    hoops.
+
+    The keytab will end up in path: <temp_working_dir>/<keytab_file>
+    """
+    log.info("Copying {} to the temp working directory".format(self.keytab_file_name))
+
+    # 1. copy from within container to private agent
+    copy_cmd = "sudo cp /var/lib/mesos/slave/slaves/{host_id}/frameworks/{framework_id}/executors/{task_id}/runs/latest/{keytab_file} /home/{linux_user}/{keytab_file}".format(
+        host_id=self.kdc_host_id,
+        framework_id=self.framework_id,
+        task_id=self.task_id,
+        keytab_file=self.keytab_file_name,
+        linux_user=LINUX_USER
+    )
+    cmd = "node ssh --master-proxy --mesos-id={host_id} --option StrictHostKeyChecking=no '{copy_cmd}'".format(
+        host_id=self.kdc_host_id, copy_cmd=copy_cmd)
+    try:
+        sdk_cmd.run_cli(cmd)
```

Comment:
@NimaVaziri did this work locally? It failed in CI with:
```
There is no SSH_AUTH_SOCK env variable, which likely means you aren't running `ssh-agent
```

The existing pattern we use is to use the Shakedown SSH utilities. @mellenburg given that Paramiko has issues often, is it possible to make a direct SSH work?

Reply:
In other tests we have: `shakedown.run_command_on_agent` which I assume encapsulates this to some extent.

Reply:
Yep, that runs via Paramiko ("native" python ssh)

Reply:
Yeah this worked locally, but saw that it failed CI. Will need to address it.

Reply:
I was able to run this locally in a docker container:
```bash
$ eval "$(ssh-agent -s)"
$ ssh-add -k /ssh/key
```
(we could try adding these ... note the `:dev` image has a different ssh key)

Reply:
The `:dev` image has a ket at `/root/.ssh/id_rsa`, which one could try adding directly to the ssh commands or having the agent running as above. Ideally, we need to "play nice" with `dcos-launch` clusters that generate the keys on the fly, but we could address this as a next step.